### PR TITLE
Nxcm 4153 expose content authc plugin muck

### PR DESCRIPTION
**Pull request by jdillon**
Initial changes to allow control over authc for /content muck.

If this looks sane then I think this is ready to be merged if blessed.

### CODE REVIEW

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusContentRestrictedToken.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * {@link AuthenticationToken} used when access to {code}/content{code} is restricted.
+ *
+ * @see NexusContentAuthenticationFilter
+ *
+ * @since 2.1
+ */
+public class NexusContentRestrictedToken
+    implements /*RememberMeAuthenticationToken,*/ HostAuthenticationToken
```
Comment (cstamas) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusContentRestrictedToken.java:35]: IMO, no, as token acces is meant for "automated clients" (like Maven is), no need for remember me stuff

**Comment by cstamas** [comment]
:beer:

**Comment by nabcos** [comment]
+1

**Comment by mcculls** [comment]
+1

**Pull request merged**